docs/service-canary-user-manual.md

```diff
@@ -0,0 +1,169 @@
+
+```yaml
+apiVersion: mesh.megaease.com/v1alpha1
+kind: ServiceCanary
+metadata:
+  name: restaurant-mesh-beijing
+spec:
+  priority: 4
+  selector:
+    matchServices: [restaurant-mesh]
+    matchInstanceLabels: {release: restaurant-mesh-beijing}
+  trafficRules:
+    headers:
+      X-Location:
+        exact: Beijing
+```
+
+The lower number is, the priority is higher. So the traffic from Beijing will be tagged `restaurant-mesh-beijing` instead of `delivery-mesh-beijing`. So if the priority of `restaurant-mesh-beijing` was 6, `delivery-mesh-beijing` will be the one.
+
+Multiple canaries with the same priority will be sorted alphabetically in matching, but the user had better rely on the priority instead of the name in order to get explicit results.
+
+So after understanding the mechanism, we could apply the config:
+
+```bash
+$ emctl apply -f https://raw.githubusercontent.com/megaease/easemesh-demo/main/deploy/mesh/easemesh_restaurant_beijing.yaml
+$ kubectl apply -f https://raw.githubusercontent.com/megaease/easemesh-demo/main/deploy/mesh/k8s_restaurant_beijing.yaml
+
+$ curl http://127.0.0.1:32539/ -d '{"order_id": "abc1234", "food": "bread"}' -H 'X-Location: Beijing'
+order_id: abc1234
+restaurant:
+  delivery_time: '2021-12-07T15:11:33 (cook duration: 5m)'
+  food: bread
+  order_id: abc1234
+```
+
+Now Beijing traffic will go through the new restaurant canary, which doesn't get mixed with delivery canary without confusion. What if you want it to go through delivery canary, adding the header `X-Mesh-Service-Canary: delivery-mesh-beijing` will get the last result as expected.
+
+## Service Canary Across Multiple Services
+
+![image](./imgs/service-canary-03.png)
+
+So what if delivery and restaurant need to be tested for a new feature together. We prepare a feature that restaurant returns coupon if delivery returns the delivery time is beyond the deadline.
+
+```bash
+$ emctl apply -f https://github.com/megaease/easemesh-demo/raw/main/deploy/mesh/easemesh_android.yaml
+$ kubectl apply -f https://raw.githubusercontent.com/megaease/easemesh-demo/main/deploy/mesh/k8s_delivery_android.yaml
+$ kubectl apply -f https://raw.githubusercontent.com/megaease/easemesh-demo/main/deploy/mesh/k8s_restaurant_android.yaml
+
+curl http://127.0.0.1:32539/ -d '{"order_id": "abc1234", "food": "bread"}' -H 'X-Phone-Os: Android'
+order_id: abc1234
+restaurant:
+  coupon: $5
+  delivery_time: 2021-12-07T16:54:01
+  food: bread
+  order_id: abc1234
+```
+
+The config shows it very clearly:
+
+```yaml
+apiVersion: mesh.megaease.com/v1alpha1
+kind: ServiceCanary
+metadata:
+  name: refund-android
+spec:
+  priority: 5
+  selector:
+    matchServices: [restaurant-mesh, delivery-mesh]
+    matchInstanceLabels: {release: refund-android}
+  trafficRules:
+    headers:
+      X-Phone-Os:
+        exact: Android
+```
+
```

Comment:
The user manual is a reference, you need to give users the configuration details. Suggesting refer to [here](https://github.com/megaease/easemesh-api/blob/main/v1alpha1/meshmodel.md#easemesh.v1alpha1.ServiceCanary)